### ISSUE TITLE
Adding locked-out data layer?

### ISSUE THREAD

**Issue by sandra-neubert** [ELSA-UNDP/elsar#27]
Currently, the data is being filtered based on the HFP in a pre-processing step to create the zones, which is happening quite hidden from the user and makes any downstream changes difficult if some of the methodology changes/ is adapted at some point. It might also be difficult to understand why the output looks a certain way (e.g. clumped). An alternative would be to create locked-out layers that would define areas that are locked out of the optional PUs for each zone with a constraint (so the opposite of a locked-in area; https://prioritizr.net/reference/add_locked_out_constraints.html), making the steps a bit more transparent because it can be directly integrated into the prioritizr analysis step, and it could be easily adapted later on. It also allows for a very easy and straightforward comparison of results with and without filter (locked-out yes or no). Thoughts @scabecks ?

**Comment by scabecks** [comment]
We should have a chat with Oscar about this. If it were simply a question of locking out PAs/OECMs from the restore/manage/urban greening zones (or soemthing similar) then I think that is a useful simplifications in the method and transparent. If we intent to (pre) induce clumping without adding a bpf then it would still require making some layer that represents those clumped locked out areas though, right?

**Comment by sandra-neubert** [comment]
Yes, I was thinking more of the second part of what you were saying: create a layer that includes the locked-out areas and add that as a locked-out constraint in the prioritizr analysis. It might make it more transparent than changing the zones input before even putting it into the prioritization and constraints usually don't increase the prioritizr run time majorly, so run time shouldn't be an issue @scabecks.
